dll/win32/shell32/dialogs/item_prop.cpp

```diff
@@ -0,0 +1,182 @@
+/*
+ * PROJECT:     shell32
+ * LICENSE:     LGPL-2.1+ (https://spdx.org/licenses/LGPL-2.1+)
+ * PURPOSE:     FileSystem PropertySheet implementation
+ * COPYRIGHT:   Copyright 2024 Whindmar Saksit <whindsaks@proton.me>
+ */
+
+#include "precomp.h"
+
+WINE_DEFAULT_DEBUG_CHANNEL(shell);
+
+static HRESULT
+SHELL_GetCaptionFromDataObject(IDataObject *pDO, LPWSTR Buf, UINT cchBuf)
+{
+    HRESULT hr = E_INVALIDARG;
+    if (PIDLIST_ABSOLUTE pidl = SHELL_DataObject_ILCloneFullItem(pDO, 0))
+    {
+        hr = SHGetNameAndFlagsW(pidl, SHGDN_INFOLDER, Buf, cchBuf, NULL);
+        ILFree(pidl);
+    }
+    return hr;
+}
+
+struct ShellPropSheetDialog
+{
+    typedef void (CALLBACK*PFNINITIALIZE)(LPCWSTR InitString, IDataObject *pDO,
+                                          HKEY *hKeys, UINT *cKeys);
+
+    static HRESULT Show(const CLSID *pClsidDefault, IDataObject *pDO,
+                        PFNINITIALIZE InitFunc, LPCWSTR InitString)
+    {
+        HRESULT hr;
+        CRegKeyHandleArray keys;
+        if (InitFunc)
+            InitFunc(InitString, pDO, keys, keys);
+        WCHAR szCaption[MAX_PATH], *pszCaption = NULL;
+        if (SUCCEEDED(SHELL_GetCaptionFromDataObject(pDO, szCaption, _countof(szCaption))))
+            pszCaption = szCaption;
+        hr = SHOpenPropSheetW(pszCaption, keys, keys, pClsidDefault, pDO, NULL, NULL) ? S_OK : E_FAIL;
+        return hr;
+    }
+
+    struct DATA
+    {
+        PFNINITIALIZE InitFunc;
+        LPWSTR InitString;
+        CLSID ClsidDefault;
+        const CLSID *pClsidDefault;
+        IStream *pObjStream;
+    };
+
+    static void FreeData(DATA *pData)
+    {
+        if (pData->InitString)
+            SHFree(pData->InitString);
+        SHFree(pData);
+    }
+
+    static HRESULT ShowAsync(const CLSID *pClsidDefault, IDataObject *pDO,
+                             PFNINITIALIZE InitFunc, LPCWSTR InitString)
+    {
+        DATA *pData = (DATA*)SHAlloc(sizeof(*pData));
+        if (!pData)
+            return E_OUTOFMEMORY;
+        ZeroMemory(pData, sizeof(*pData));
+        pData->InitFunc = InitFunc;
+        if (InitString && FAILED(SHStrDupW(InitString, &pData->InitString)))
+        {
+            FreeData(pData);
+            return E_OUTOFMEMORY;
+        }
+        if (pClsidDefault)
+        {
+            pData->ClsidDefault = *pClsidDefault;
+            pData->pClsidDefault = &pData->ClsidDefault;
+        }
+
+        HRESULT hr = S_OK;
+        if (pDO)
+            hr = CoMarshalInterThreadInterfaceInStream(IID_IDataObject, pDO, &pData->pObjStream);
+
+        UINT flags = CTF_COINIT | CTF_PROCESS_REF | CTF_INSIST;
```

Comment:
```suggestion
        const UINT flags = CTF_COINIT | CTF_PROCESS_REF | CTF_INSIST;
```

Reply:
This has ZERO impact on the generated code but OK.